### PR TITLE
new postcode lookup endpoint changes

### PR DESCRIPTION
**Pull request by mrganeshraja**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-4890

### Change description ###
Changes to new postcode lookup service

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: app/address/address-lookup.js [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 
 const logger = Logger.getLogger('addressLookup');
 
 
-function addressLookup(postcode) {
+function addressLookup() {
 
   return fetch(
-    config.address_lookup.url.replace('${postcode}', postcode), getHttpConfig())
+    config.address_lookup.url.replace('${key}', config.get('secrets.ccd.postcode-info-address-lookup-token')), getHttpConfig())
```
Comment (andrewfolga) [inline app/address/address-lookup.js]: Where is the `key` coming from here?

File: test/address/address-lookup.spec.js [hunk 2]
```diff
@@ -2,136 +2,65 @@ const chai = require('chai');
...
-          expect(addresses[1]).to.equal(extractedAddress3);
-          expect(addresses[2]).to.equal(extractedAddress1);
-          done();
-        }
-        catch (e) {
-          done(e);
-        }
-      });
-   });
+  it('Should expect key to replaced and not the postcode', (done) => {
```
Comment (andrewfolga) [inline test/address/address-lookup.spec.js]: key to `be` replaced

File: app/address/address-lookup.js (hunk 1 above)
Comment (andrewfolga) [inline app/address/address-lookup.js]: We do it here as well as in the `ccd-case-management-web`?

File: test/address/address-lookup.spec.js (hunk 2 above)
Comment (mrganeshraja) [reply test/address/address-lookup.spec.js]: In this test the key will be replaced from default config value 'AAA' and in the non-prod and prod environment value will come through ccd-vault.

File: app/address/address-lookup.js (hunk 1 above)
Comment (mrganeshraja) [reply app/address/address-lookup.js]: ccd-case-management-web will have to postcode lookup url to gateway configuration address and that will get passed postcode lookup component and the component will call the gateway URL with the postcode attribute and gateway will fill the api key value.

Comment (andrewfolga) [reply app/address/address-lookup.js]: Ok now I see it is update in man web 👍

File: app/address/address-lookup.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 const config = require('config');
-const extractAddress = require('./extract-address');
+// const extractAddress = require('./extract-address');
```
Comment (mario-paniccia) [inline app/address/address-lookup.js]: can we remove this?

File: config/default.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ idam:
     client_id: ccd_gateway
 address_lookup:
   detect_proxy: false
-  url: https://postcodeinfo.service.justice.gov.uk/addresses?postcode=${postcode}
+  url:  https://api.ordnancesurvey.co.uk/places/v1/addresses/postcode?postcode=${postcode}&key=${key}
```
Comment (mario-paniccia) [inline config/default.yaml]: can you remove the extra space please

Comment (mrganeshraja) [reply config/default.yaml]: 👍

File: app/address/address-lookup.js (hunk 3 above)
Comment (mrganeshraja) [reply app/address/address-lookup.js]: 👍

**Pull request merged**